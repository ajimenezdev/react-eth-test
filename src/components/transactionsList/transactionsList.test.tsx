import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render } from "@testing-library/react";
import { act } from "@testing-library/react-hooks";
import TransactionLists from "./index";

it("renders without crashing when no arguments", () => {
  const component = renderer.create(
    <TransactionLists
      transactions={[]}
      address=""
      network=""
      updating={false}
      onAddressClick={() => null}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders progress when updating=true", () => {
  act(() => {
    const { getByLabelText } = render(
      <TransactionLists
        transactions={[]}
        address=""
        network=""
        updating={true}
        onAddressClick={() => null}
      />
    );

    expect(getByLabelText("loading")).toBeVisible();
  });
});

it("renders items when transaction is not empty", () => {
  const { getAllByLabelText } = render(
    <TransactionLists
      transactions={[
        {
          nonce: 1,
          hash: "hash1",
          from: "address1",
          to: "address2",
          value: 10
        },
        {
          nonce: 2,
          hash: "hash2",
          from: "address1",
          to: "address2",
          value: 10000
        }
      ]}
      address="address1"
      network="Mainnet"
      updating={false}
      onAddressClick={() => null}
    />
  );

  expect(getAllByLabelText("list-tx-item"));
});

it("triggers callback when address clicked", () => {
  const mockCallback = jest.fn();
  const { getAllByLabelText } = render(
    <TransactionLists
      transactions={[
        {
          nonce: 1,
          hash: "hash1",
          from: "address1",
          to: "address2",
          value: 10
        },
        {
          nonce: 2,
          hash: "hash2",
          from: "address1",
          to: "address2",
          value: 10000
        }
      ]}
      address="address1"
      network="Mainnet"
      updating={false}
      onAddressClick={mockCallback}
    />
  );

  const address = getAllByLabelText("address_link")[0];
  act(() => {
    fireEvent.click(address);
  });
  expect(mockCallback.mock.calls.length).toBe(1);
});
