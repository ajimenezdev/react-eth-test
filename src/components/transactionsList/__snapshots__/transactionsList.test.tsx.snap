// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing when no arguments 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 body-list MuiPaper-rounded"
>
  <div
    className="list-filters"
  >
    <div
      className="list-filterItem"
    >
      <span>
        Select Transaction Source:
      </span>
      <div
        className="MuiToggleButtonGroup-root"
        role="group"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="both"
        >
          <span
            className="MuiToggleButton-label"
          >
            Both
          </span>
        </button>
        <button
          aria-pressed={true}
          className="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="from"
        >
          <span
            className="MuiToggleButton-label"
          >
            From
          </span>
        </button>
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="to"
        >
          <span
            className="MuiToggleButton-label"
          >
            To
          </span>
        </button>
      </div>
    </div>
    <div
      className="list-filterItem"
    >
      <span>
        Show last:
      </span>
      <div
        className="MuiFormControl-root MuiTextField-root list-countFilter MuiFormControl-marginNormal"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            10
          </div>
          <input
            type="hidden"
            value={10}
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <ul
    className="list"
  >
    <li
      className="list-item list-header"
    >
      <span
        className="list-nonce"
      >
        Nonce
      </span>
      <span
        className="list-hash"
      >
        T.Hash
      </span>
      <span
        className="list-from"
      >
        From
      </span>
      <span
        className="list-to"
      >
        To
      </span>
      <span
        className="list-value"
      >
        Value
      </span>
    </li>
    <p>
      There are not transactions
    </p>
  </ul>
</div>
`;
